js-packages/@prestojs/rest/src/usePaginator.ts

```diff
@@ -13,9 +13,26 @@ import { PaginatorInterface, PaginatorInterfaceClass } from './Paginator';
  * is handled internally in usePaginator. You can pass `useState()` to this parameter.
  */
 export default function usePaginator<T extends PaginatorInterface, PaginatorState>(
-    paginatorClassOrEndpoint: PaginatorInterfaceClass<T> | Endpoint<T>,
+    paginatorClassOrEndpoint: PaginatorInterfaceClass<T> | Endpoint<any, T>,
     currentStatePair?: [PaginatorState | undefined, (value: PaginatorState) => void]
-): T {
+): T;
+export default function usePaginator<T extends PaginatorInterface, PaginatorState>(
+    paginatorClassOrEndpoint: null,
+    currentStatePair?: [PaginatorState | undefined, (value: PaginatorState) => void]
+): null;
+// This is only necessary because the type is inferred as PaginatorInterfaceClass<T> | null here
+// instead of refining it based on endpoint
+// export function ExampleComponent({ endpoint }: { endpoint: Endpoint | null }) {
+//     usePaginator(endpoint && endpoint.getPaginatorClass());
+// }
```

Comment:
Does this need a test case or is it simple enough that it's not worth it?

Reply:
test added